File: day1/part1.mjs
import {readFileSync} from "fs";

const input = readFileSync("./input.txt").toString();

var firstDigit = -1;
var lastDigit = -1;
var total = 0;
for (var i = 0; i < input.length; i++) {
    var charCode = input.charCodeAt(i);
    if (charCode == 10) {
        if (firstDigit != -1) {
            if (lastDigit != -1) {
                console.log("Two digits on line", firstDigit, lastDigit);
                total += (10 * firstDigit);
                total += lastDigit;
                firstDigit = -1;
                lastDigit = -1;
            } else {
                console.log("One digit on line", firstDigit);
                total += 11 * firstDigit;
                firstDigit = -1;
                lastDigit = -1;
            }
        } else {
            console.log("No digits on this line");
        }
    }
    if (charCode >= 48 && charCode <= 57) {
        console.log("digit", input[i], charCode);
        if (firstDigit == -1) {
            firstDigit = charCode - 48;
            console.log("First digit", input[i], charCode - 48);
        } else {
            lastDigit = charCode - 48;
            console.log("Last digit", input[i], charCode - 48);
        }
    } else {
        console.log("not digit", input[i], charCode);
    }
}
console.log("Total", total);

File: day1/part1.c
#include <stdio.h>

int getLineTotal(int a);

int main() {
    //FILE *fptr;

    // Open a file in read mode
    //fptr = fopen("input.txt", "r");

    //char buf[200];
    //int firstDigit = -1;
    //int lastDigit = -1;
    int total = 0;
    //while (fgets(buf, sizeof buf, fptr) != NULL) {
    //    total += getLineTotal(buf);
    //}

    int i = 0;
    while (i < 6) {
        total = getLineTotal(i);
        i++;
    }
          
    printf("Total %i\n", total);

    //if (feof(fptr)) {
    //    printf("Total %i", total);
    //}

    // Close the file
    //fclose(fptr);

    return 0;
}

int getLineTotal(int a) {
    return a + 1;
}

/* int getLineTotal(char *buf) {
    int firstDigit = -1;
    int lastDigit = -1;
    int lineTotal = 0;
    //printf("%s", buf);
    int i = 0;
    while (buf[i] != 0) {
        char c = buf[i];
        if (c >= 48 && c <= 57) {
            if (firstDigit == -1) {
                firstDigit = c - 48;
                //printf("FD: %i - ", firstDigit);
            } else {
                lastDigit = c - 48;
                //printf("LD: %i - ", lastDigit);
            }
        }
        i++;
    }
    if (lastDigit == -1) {
        lineTotal = 11 * firstDigit;
    } else {
        lineTotal += firstDigit * 10;
        lineTotal += lastDigit;
    }
    return lineTotal;
} */